lib/vector-core/src/event/metric.rs

```diff
@@ -223,6 +224,53 @@ impl MetricValue {
             Self::Sketch { sketch } => sketch.as_name(),
         }
     }
+
+    /// Converts a distribution to an aggregated histogram.
+    ///
+    /// Histogram bucket bounds are based on `buckets`, where the value is the upper bound of the
+    /// bucket.  Samples will be thus be ordered in a "less than" fashion: if the given sample is
+    /// less than or equal to a given bucket's upper bound, it will be counted towards that bucket
+    /// at the given sample rate.
+    ///
+    /// If this `MetricValue` is not a distribution, then `None` is returned.  Otherwise,
+    /// `Some(MetricValue::AggregatedHistogram)` is returned.
+    pub fn distribution_to_agg_histogram(self, buckets: &[f64]) -> Option<MetricValue> {
```

Comment:
Yeah, that's fair: we don't actually consume/take ownership of the distribution samples anyways since they have to be transformed one way or another.

Switched these to take `&self` in 9a08c0198, which required no other changes, surprisingly.